pages/index.js

```diff
@@ -1,77 +1,160 @@
-import Head from "next/head";
-import { useState, useEffect } from "react";
-import styled from "styled-components";
-import Button from "@/components/ui/Button";
+import ActivityCard from "@/components/layout/ActivityCard";
 import Header from "@/components/layout/Header";
-import ActivityList from "@/components/layout/ActivityList";
-import ActivityForm from "@/components/layout/ActivityForm";
-import ActivityFilter from "@/components/layout/ActivityFilter";
-
-export default function HomePage({
-  handleAddActivity,
-  activities,
-  bookmarks,
-  toggleBookmark,
-  handleFilter,
-  filter,
-}) {
-  const [showForm, setShowForm] = useState(false);
-  const [showFilter, setShowFilter] = useState(false);
-
-  const activity = {
-    id: "",
-    title: "",
-    categories: [],
-    area: "",
-    country: "",
-    description: "",
-    imageUrl: "",
-  };
-
-  function handleToggleEdit() {
-    setShowForm(!showForm);
+import Button from "@/components/ui/Button";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+import { FaSearch } from "react-icons/fa";
+
+export default function ActivityPage({ activities, toggleBookmark,bookmarks,deleteActivity,showHeart }) {
+  const [randomActivities, setRandomActivities] = useState([]);
+  const NUM_OF_RANDOM_ACTIVITIES = 6;
+
+  useEffect(() => {
```

Comment:
You're already passing activities as a prop and generating `randomActivities` inside the component, which means it will automatically update whenever the activities prop changes. This makes useEffect redundant. Consider a solution without useEffect